spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/prometheus/PrometheusMetricsExportAutoConfiguration.java

```diff
@@ -117,13 +119,22 @@ public PrometheusPushGatewayManager prometheusPushGatewayManager(
 				PrometheusProperties prometheusProperties, Environment environment) {
 			PrometheusProperties.Pushgateway properties = prometheusProperties
 					.getPushgateway();
-			PushGateway pushGateway = new PushGateway(properties.getBaseUrl());
 			Duration pushRate = properties.getPushRate();
 			String job = getJob(properties, environment);
 			Map<String, String> groupingKey = properties.getGroupingKey();
 			ShutdownOperation shutdownOperation = properties.getShutdownOperation();
-			return new PrometheusPushGatewayManager(pushGateway, collectorRegistry,
-					pushRate, job, groupingKey, shutdownOperation);
+			return new PrometheusPushGatewayManager(
```

Comment:
I don't see a reason to change the flow of the code here. I also feel that the previous version was easier to read. Can you please revert that part?

Reply:
`spring-java-format` :(
I've introduced a variable to avoid this.

Reply:
That's probably wasn't what I was talking about. I meant the fact you remove the `PushGateway` variable and therefore changed this constructor. I see no good reason for doing so.

Reply:
There are 2 constructors.
 - `String`  creates `URL` as `"http://" + address` and delegates it to underlying `URL` constructor.
- `URL` uses `URL` as is.

So, that is why I've introduced a method to create a `PushGateway` instance.

Reply:
That's not what I am talking about and what you pushed force in the meantime is addressing it, thank you.